app.py

```diff
@@ -60,17 +95,70 @@ def show_in_filesystem(self, entity_type, entity_ids):
         with the given entity ids.
         """
         paths = []
-        
+
+        # As the this method relies on the path_cache to find the folders,
+        # we must check we are in sync before continuing
+        self.sgtk.synchronize_filesystem_structure()
+
         for eid in entity_ids:
+
+            if entity_type == "PublishedFile":
+                # Find the publish so that we can get the path back.
+                published_file = self.sgtk.shotgun.find_one("PublishedFile", [["id", "is", eid]], ['code', 'path'])
+                # Resolve the path for the local OS
+                try:
+                    local_path = sgtk.util.resolve_publish_path(self.sgtk, published_file)
+                except Exception as e:
+                    self.log_debug("Publish path couldn't be resolved falling back to context based path; reason: %s" % e)
```

Comment:
I think this is a good point. My approach with adding the changes was to always fall back on the current behaviour (which is to use the context paths) if for whatever reason we could not find a better more specific way of showing the paths.
Perhaps your right, now it supports `PublishedFile` entities it should just fail if it can't find the path?